src/compressor/Compressor.cc

```diff
@@ -23,40 +23,21 @@
 #include "common/dout.h"
 
 const char * Compressor::get_comp_alg_name(int a) {
-  switch (a) {
-  case COMP_ALG_NONE: return "none";
-  case COMP_ALG_SNAPPY: return "snappy";
-  case COMP_ALG_ZLIB: return "zlib";
-  case COMP_ALG_ZSTD: return "zstd";
-#ifdef HAVE_LZ4
-  case COMP_ALG_LZ4: return "lz4";
-#endif
-#ifdef HAVE_BROTLI
-  case COMP_ALG_BROTLI: return "brotli";
-#endif
-  default: return "???";
-  }
+
+  if(a < 0 || a >= COMP_ALG_LAST) 
+   return "???"; // It would be nice to revise this...
+
+  return compression_algorithm_names[a];
 }
 
 boost::optional<Compressor::CompressionAlgorithm> Compressor::get_comp_alg_type(const std::string &s) {
-  if (s == "snappy")
-    return COMP_ALG_SNAPPY;
-  if (s == "zlib")
-    return COMP_ALG_ZLIB;
-  if (s == "zstd")
-    return COMP_ALG_ZSTD;
-#ifdef HAVE_LZ4
-  if (s == "lz4")
-    return COMP_ALG_LZ4;
-#endif
-#ifdef HAVE_BROTLI
-  if (s == "brotli")
-    return COMP_ALG_BROTLI;
-#endif
-  if (s == "" || s == "none")
-    return COMP_ALG_NONE;
 
-  return boost::optional<CompressionAlgorithm>();
+  for (int pos = 0; pos != sizeof(compression_algorithm_names); ++pos) {
```

Comment:
Why != instead of <= in this loop? I thought that was generally considered bad because if the loop mutates and you accidentally skip over the end condition, you're in trouble...

Reply:
@gregsfortytwo Should be < instead of <=

Reply:
If a static array mutates, you're in real trouble! ;-)